### PR TITLE
Feat: Support for Question Papers (including Blueprint creation/progress measurement)

### PR DESCRIPTION
**Pull request by bharatkashyap**
Add Blueprint, Add Tags to Exam Questions against Blueprint, View Blueprint, View Progress Against Blueprint

### CODE REVIEW

File: src/app/client/src/app/modules/sourcing/components/content-editor/content-editor.component.ts [hunk 1]
```diff
@@ -131,6 +131,7 @@ export class ContentEditorComponent implements OnInit, OnDestroy, AfterViewInit
       this.contentStatusNotify(action);
     });
     this.helperService.initialize(this.programContext);
+    debugger;
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-editor/content-editor.component.ts:134]: Please Remove the debugger.

File: src/app/client/src/app/modules/sourcing/components/bulk-upload/bulk-upload.component.ts [hunk 2]
```diff
@@ -62,7 +62,8 @@ export class BulkUploadComponent implements OnInit {
   public bulkUploadErrorMsgs = [];
   public bulkUploadValidationError = '';
   public levels = [];
-  public sampleMetadataCsvUrl: string = (<HTMLInputElement>document.getElementById('portalCloudStorageUrl')).value.split(',')  + 'bulk-content-upload-format.csv';
+  // public sampleMetadataCsvUrl: string = (<HTMLInputElement>document.getElementById('portalCloudStorageUrl')).value.split(',')  + 'bulk-content-upload-format.csv';
+  public sampleMetadataCsvUrl: string = 'bulk-content-upload-format.csv';
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/bulk-upload/bulk-upload.component.ts:66]: @pallakartheekreddy please check this.

**Review by venkateshwarans — COMMENTED** [review]
Check Comments. Looks good otherwise.

File: src/app/client/src/app/modules/sourcing/components/bulk-upload/bulk-upload.component.ts (hunk 2 above)
Comment (bharatkashyap) [reply src/app/client/src/app/modules/sourcing/components/bulk-upload/bulk-upload.component.ts:66]: @venkateshwarans Sorry, this is my bad! Will correct and update.

**Comment by bharatkashyap** [comment]
> Check Comments. Looks good otherwise.

@venkateshwarans This is ready to be merged since Friday - any updates?

**Comment by pallakartheekreddy** [comment]
@bharatkashyap Can you please add unit test cases for the code you have written.

**Comment by bharatkashyap** [comment]
> @bharatkashyap Can you please add unit test cases for the code you have written.

Hi @pallakartheekreddy - got it! Do you have any specific changes in mind that need additional test cases to be added?

**Comment by bharatkashyap** [comment]
> @bharatkashyap Can you please add unit test cases for the code you have written.

The latest commit addresses this @pallakartheekreddy

**Comment by pallakartheekreddy** [comment]
@bharatkashyap I have merged the PR. Please do one round of test from you and release the tickets to QA team and next time on words please add JIRA ID's also in the PR.

**Comment by bharatkashyap** [comment]
> @bharatkashyap I have merged the PR. Please do one round of test from you and release the tickets to QA team and next time on words please add JIRA ID's also in the PR.

Noted @pallakartheekreddy - will make sure JIRA IDs are present next time, and will update this one with the JIRA IDs as well.

**Pull request merged**